include/deal.II/dofs/dof_tools.h

```diff
@@ -2260,6 +2275,33 @@ namespace DoFTools
   std::vector<Table<2,Coupling> >
   dof_couplings_from_component_couplings (const hp::FECollection<dim,spacedim> &fe,
                                           const Table<2,Coupling> &component_couplings);
+
+  /**
+   * Given a finite element and a table how the vector components of it couple
+   * with each other, compute and return a table that describes how the
+   * individual shape functions couple with each other, in boolean form rather
+   * than Coupling form.
+   *
+   * The function currently treats DoFTools::Couplings::nonzero the same as
+   * DoFTools::Couplings::always.
+   */
+  template <int dim, int spacedim>
+  Table<2,bool>
+  bool_dof_couplings_from_component_couplings (const FiniteElement<dim,spacedim> &fe,
+                                               const Table<2,Coupling> &mask_in);
```

Comment:
I don't think I quite understand the description or purpose of this function. Is this function essentially "flattening" the `Table<2,Couplings>` to `Table<2,bool>` by converting each entry to a boolean by comparing `!= none`?